packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -206,8 +206,11 @@ class AndroidSdk {
     if (!processManager.canRun(adbPath))
       return <String>['Android SDK file not found: $adbPath.'];
 
-    if (sdkVersions.isEmpty || latestVersion == null)
-      return <String>['Android SDK is missing command line tools; download from https://goo.gl/XxQghQ'];
+    if (sdkVersions.isEmpty)
+      return <String>['No Android SDK version found (check SDK for standard layout).'];
```

Comment:
What does "standard layout" mean? And what does it mean to check it?
- How does the user verify that the SDK layout is standard?
- What should the user do, if it is not?

Reply:
I think it's clearer to just say where we looked for the versions and what we found.  Something like:

```dart
Iterable<Directory> platforms = <Directory>[]; // android-22, ...
final Directory platformsDir = fs.directory(fs.path.join(directory, 'platforms'));
if (platformsDir.existsSync()) {
  platforms = platformsDir
    .listSync()
    .where((FileSystemEntity entity) => entity is Directory)
    .map<Directory>((FileSystemEntity entity) {
      final Directory dir = entity;
      return dir;
    });
}

StringBuffer msg = new StringBuffer('No valid Android SDK platforms found in ${platformsDir.path}.');
if (platforms.isEmpty) {
  msg.write('  Directory was empty.');
} else {
  msg.write('Candidates were:\n');
  msg.write(platforms
      .map((Directory dir) => '  - ${dir.basename}')
      .join('\n'));
}
return <String>[msg.toString()];
```

Note that the first paragraph above already exists in `_init()` and should be refactored out into `get _platforms` and `get _platformsDir`.